Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -225,6 +223,12 @@ inline WICPixelFormatGUID PixelFormat() const {
         _impl.renderingIntent = intent;
         return *this;
     }
+
+    ~__CGImage() {
+        for (auto listener: _imageDestructionListeners) {
+            listener(this);
+        }
```

Comment:
do we need a lock around this?  also, kind of odd to have an instance free all static listeners....are you sure this is working correctly?